addons/road-generator/road_point.gd

```diff
@@ -286,7 +282,14 @@ func assign_lanes():
 
 		# Now complete the final lane.
 		if running_same_dir > 0:
-			lanes.append(LaneType.SLOW)
+			if running_same_dir == len(flips):
+				if traffic_dir[-1] == LaneDir.FORWARD:
+					lanes.append(LaneType.SLOW)
+					lanes[0] = LaneType.NO_MARKING
+				else:
```

Comment:
I'll also say I'm weary about this level of code indention, gets to be rather hard to read. But I'm not sure there's a worthwhile improvement in the immediate term. Unless you can think of a way to pivot more into a same-level nesting case statement? Or precompute the combined logical checks so it can be more like `if .. elif ... elif ... elif ... ... else`?

Reply:
This is a nice to have if you can get to it, but I'll approve in the meantime.

Reply:
> Doing some reading, we essentially need to perform a bezier offset - but that so far really just means taking samples of an existing bezier curve, with spacing based on the rate of curvature. This can get us a new curve in the offset position which will have many points, more than the two points of the source, but more closely match the curve in question.

Yeah, it might be nice to have a Segment (or RoadPoint) property that says whether to apply the curve to the left/center/right of the Segment.